tutorials/international/index.md

```diff
@@ -0,0 +1,281 @@
+[`MaterialApp`](https://docs.flutter.io/flutter/material/MaterialApp-class.html)
+(or its [`WidgetsApp`](https://docs.flutter.io/flutter/widgets/WidgetsApp-class.html)
+for non-material applications):
+
+```dart
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      localizationsDelegates: <LocalizationsDelegate>[
+        new MyLocalizationsDelegate()
+      ],
+      supportedLocales: <Locale>[
+        const Locale('en', 'US'), // US English
+        const Locale('fr', 'CA'), // Candian French
+      ],
+      child: home: new Scaffold(
+        appBar: new AppBar(
+          title: new Text(MyLocalizations.of(context).helloWorld),
+          ...
+        ),
+      ),
+    );
+  }
+}
+```
+
+As you can see, the app's `AppBar` title comes from an instance of
+`MyLocalizations` produced by `MyLocalizations.load()` via
+`MyLocalizationsDelegate`. If the device's locale changes, then the
+app will be rebuilt with a new `MyLocalizations` object, one
+constructed for the new locale. Any reference to
+`MyLocalizations.of()` anywhere in the app will causes its context to
+be rebuilt if the locale changes.
+
+### The App's `supportedLocales` Parameter
+
+Although the Flutter's Material library includes support for about 16
+languages, only English language translations are available by
+default. It's up to applications to decide exactly which languages
+they'll support, since it wouldn't make sense for the toolkit
+libraries to support a different set of locales than the app does.
+
+The `MaterialApp`
+[`supportedLocales`](https://docs.flutter.io/flutter/material/MaterialApp-class.html)
+parameter limits locale changes. When the user changes the locale
+setting on their device, the app's `Localizations` widget will only
+follow suit if the new locale is a member of the this list. If an
+exact match for the device locale isn't found, then the first
+supported locale with a matching
+[`languageCode`](https://docs.flutter.io/flutter/dart-ui/Locale/languageCode.html)
+is used. If that fails, then the first element of the
+`supportedLocales` list is used.
+
+In terms of the previous `MyApp` example, the app will only accept the
+US English or French Canadian locales and it will substitute US
+English (the first locale in the list) for anything else.
+
+An app that wants to use a different "locale resolution" method, can
+provide a
+[`localeResolutionCallback`](https://docs.flutter.io/flutter/material/MaterialApp-class.html). For
+example to have your app unconditionally accept whatever locale the
+user selects:
+
+```dart
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+       localeResolutionCallback(Locale locale, Iterable<Locale> supportedLocales) {
+         return locale;
+       }
+       // ...
+    );
+  }
+}
+
+```
+
+
+### An Alternative Class for the App's Localized Resources
```

Comment:
Curious, will this be confusing to developers? What's our opinion? If our opinion is to use intl, we might not need this section...

Reply:
The feedback I got from an earlier version of this doc was that a non-intl package example would be helpful. Our opinion is that you can use intl if you want.